components/cam/src/dynamics/se/inidat.F90

```diff
@@ -453,31 +453,32 @@ subroutine read_inidat( ncid_ini, ncid_topo, dyn_in)
 
     if ( (ideal_phys .or. aqua_planet)) then
        tmp(:,1,:) = 0._r8
-       phis_tmp(:,:) = 0._r8
+       if (fv_nphys > 0) phis_tmp(:,:) = 0._r8
```

Comment:
Can you elaborate on this a bit? How come prescribing no topography applies only for pg grid cases when you are simulating ideal_phys or aqua planet?

Reply:
The flat topography is prescribed in both cases using either "tmp" or "phis_tmp". These are loading similar data, but they have different dimensions. "tmp" is just a general variable for loading data on to the np4 grid (second dim is nlev), whereas "phis_tmp" is only for loading the topo data on the physgrid. I could have made "tmp" work in this case, but I figured it was clearer to use a new variable.

Reply:
oh I see. yes I can see that up in line 132 or so, phis_tmp is allocated only for pg grids, so you would need that if statement to go with phys_tmp.